partition_manager: Add PM_EXTERNAL_FLASH_HAS_DRIVER Kconfig option (#8515)
nrfconnect/sdk-nrf: scripts/partition_manager/partition_manager.rst

.. _pm_build_system:

> For a partition to be accessible at run-time, a flash device it is placed on needs to have driver

Reviewer: ```suggestion
For a partition to be accessible at run-time, a flash device it is placed on needs to have its driver
```

> enabled at compile time. Partition manager will not included, into partition map, partitions that

Reviewer: ```suggestion
enabled at compile time. Partition manager will not include, into partition map, partitions that
```

> are placed on a device without driver in the system.

Reviewer: ```suggestion
are placed on a device without a driver in the system.
```

:kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` in application configuration.

As stated before, the partition manager will only include into compilation partitions that
it is able to match with a driver, as pointed by the ``DEFAULT_DRIVER_KCONFIG``.
Partition manager does not know whether partitions are used at run-time or not, it only
knows whether they have driver compiled in or not.

This has following effects:

  * Enabling Kconfig that affect the ``DEFAULT_DRIVER_KCONFIG`` will add partition map entry
    for the partition depending on it, whether it is used in run-time or not.

  * Not enabling the Kconfig that affect the ``DEFAULT_DRIVER_KCONFIG`` will not add partition
    map entry for partition depending on it, whether it is used in run-time or not.

  * Enabling Kconfig options that affects the ``DEFAULT_DRIVER_KCONFIG`` may cause linker errors
    when the option has no effect on including driver into compilation. In such case partition
    manager adds partition map entry that has a pointer to flash device it is supposed to be
    placed on, but due to misconfiguration a driver is actually not compiled in.

>     Such situation may also be cause by setting the :kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK`

Reviewer: ```suggestion
    Such situation may also be caused by setting the :kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK`
```

    that will just assume that driver is provided by application.

.. note::

   When using app configuration with MCUboot child image, both images use the same partition
   manager configuration, which means that app and MCUboot will have exectly the same partition
   maps generated, but whether flash partitions will be accessible for them, at run-time,
   depends on application and MCUboot configurations and values they give to the Kconfig options,

>    that are given to the ``DEFAULT_DRIVER_KCONFIG`` option of pratition anager region specification.

Reviewer: ```suggestion
   that are given to the ``DEFAULT_DRIVER_KCONFIG`` option of partition manager region specification.
```
